frontend/src/pages/Confirmation/ConfirmationContainer.js

```diff
@@ -27,22 +27,40 @@ import {
   confirmConfirmation,
   executeAdditionalActions,
   storeAdditionalActions,
-  storeRequestedPermissions
+  storeRequestedPermissions,
+  setIsPermissionFetchNeeded
 } from "./actions";
 import ConfirmationDialog from "./ConfirmationDialog";
 import { applyOriginalActions, createAdditionalActions } from "./createAdditionalActions";
 import { executeOriginalActions } from "./executeOriginalActions";
 
 class ConfirmationContainer extends React.Component {
   componentDidUpdate(prevProps) {
-    // Fetch Permissions if grant/revoke permission confirmation required
-    if (
+    if (this.props.location.pathname !== prevProps.location.pathname) {
+      this.props.cancelConfirmation();
+    }
+
+    // On these actions (e.g project.close), a Permission fetch is not needed
+    const isFetchingNeeded = !this.props.originalActions.some(action => {
```

Comment:
we should move the logic to the action payload which is sent by the saga. Another option is to move the logic to the reducer - If doing so you generate a new location where code has to be changed when adding a new confirmation intent. So I suggest to use the saga to decide if a confirmation needs project/subporject/workflowitem permissions or not.

Reply:
done